pkg/operator/config/config_test.go

```diff
@@ -137,3 +139,294 @@ func TestBuildConfig(t *testing.T) {
 		})
 	}
 }
+
+// TestFullConfig tests generation of a config that may be used to collect
+// metrics from Kubernetes.
+func TestFullConfig(t *testing.T) {
```

Comment:
Split this up into unit tests. Test individual components directly. We can have an all-in-one test just to make sure different components get invoked.